qdglue/metrics/grid_based_metrics.py

```diff
@@ -0,0 +1,290 @@
+import dataclasses
+from abc import ABC, abstractmethod
+from typing import Dict, Tuple
+
+import gymnasium
+import numpy as np
+
+from qdglue.types import Feature, Fitness
+
+
+@dataclasses.dataclass
+class DiscreteArchiveMetrics:
+    """Holds statistics about an archive.
+
+    Attributes:
+        coverage: Proportion of cells in the archive that have an elite - always
+            in the range $[0,1]$.
+        qd_score: QD score, i.e. sum of objective values of all elites in the
```

Comment:
I think we should normalize this QD score only by adding the min objective in order to be consistent with the original introduction of QD score in Pugh 2016.

I think we can add another field for the version of QD score with the upper bound, but I would prefer not to call it `norm_qd_score` since in pyribs we already refer to `norm_qd_score` as the QD score I mentioned above, divided by number of archive cells.

It might be helpful to have more info before deciding on a name. How does upper bounding the fitness affect the final QD score, and does the max value of the QD score simply become the number of cells in the archive? One name that comes to mind here is `bound_norm_qd_score`, although it is a bit lengthy.

Reply:
agree we should have a new field this "normalized" QD-Score, as it has been widely used in the community (e.g. in the QD Framework paper from Cully 2018, and in the DQD paper). Also it is more interpretable than the old one from Pugh 2016. 

- I added a QD score that follows the one of Pugh 2016 and called it `qd_score_original`.
- I like `qd_score_bound_norm` for the "normalized" QD-Score, and I don't mind if it's lengthy, especially if it describes well how it is calculated ^^